lib/ansible/modules/cloud/amazon/ssm_parameter_store.py

```diff
@@ -0,0 +1,262 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'metadata_version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: ssm_parameter_store
+short_description: Manage key-value pairs in aws parameter store.
+description:
+  - Manage key-value pairs in aws parameter store.
+version_added: "2.4"
+options:
+  name:
+    description:
+      - parameter key name.
+    required: true
+  description:
+    description:
+      - parameter key desciption.
+    required: false
+  value:
+    description:
+      - Parameter value.
+    required: false
+  state:
+    description:
+      - Creates or modifies an existing parameter
+      - Deletes a parameter
+    required: false
+    choices: ['present', 'absent', 'show']
+    default: present
+  string_type:
+    description:
+      - Parameter String type
+    required: false
+    choices: ['String', 'StringList', 'SecureString']
+    default: String
+  decryption:
+    description:
+      - Work with SecureString type to get plain text secrets
+      - Boolean
+    required: false
+    default: True
+  key_id:
+    description:
+      - aws KMS key to decrypt the secrets.
+    required: false
+    default: aws/ssm (this key is automatically generated at the first parameter created).
+  overwrite:
+    description:
+      - Overwrite the value when create or update parameter
+      - Boolean
+    required: false
+    default: True
+author: Bill Wang (ozbillwang@gmail.com)
+extends_documentation_fragment: aws
+requirements: [ botocore, boto3 ]
+'''
+
+EXAMPLES = '''
+- name: Create or update key/value pair in aws parameter store
+  ssm_parameter_store:
+    name: "Hello"
+    description: "This is your first key"
+    value: "World"
+
+- name: Delete the key
+  ssm_parameter_store:
+    name: "Hello"
+    state: absent
+
+- name: Create or update secure key/value pair with default kms key (aws/ssm)
+  ssm_parameter_store:
+    name: "Hello"
+    description: "This is your first key"
+    string_type: "SecureString"
+    value: "World"
+
+- name: Create or update secure key/value pair with nominated kms key
+  ssm_parameter_store:
+    name: "Hello"
+    description: "This is your first key"
+    string_type: "SecureString"
+    key_id: "alias/demo"
+    value: "World"
+
+- name: Retrieving plain-text secret
+  ssm_parameter_store:
+    name: "Hello"
+    state: show
```

Comment:
That is indeed what I mean. It would simplify the module and eliminate the duplication of functionality across lookup plugin and module. You could refer to the lookup plugin in the module documentation though